src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Caching.cs

```diff
@@ -252,52 +257,102 @@ public static CachingContext GetOrCreate(JsonSerializerOptions options)
+                    left._numberHandling == right._numberHandling &&
+                    left._unknownTypeHandling == right._unknownTypeHandling &&
+                    left._defaultBufferSize == right._defaultBufferSize &&
+                    left._maxDepth == right._maxDepth &&
+                    left._allowTrailingCommas == right._allowTrailingCommas &&
+                    left._ignoreNullValues == right._ignoreNullValues &&
+                    left._ignoreReadOnlyProperties == right._ignoreReadOnlyProperties &&
+                    left._ignoreReadonlyFields == right._ignoreReadonlyFields &&
+                    left._includeFields == right._includeFields &&
+                    left._propertyNameCaseInsensitive == right._propertyNameCaseInsensitive &&
+                    left._writeIndented == right._writeIndented &&
+                    left._typeInfoResolver == right._typeInfoResolver &&
+                    CompareLists(left._converters, right._converters);
+
+                static bool CompareLists<TValue>(ConfigurationList<TValue> left, ConfigurationList<TValue> right)
                 {
-                    return false;
+                    int n;
+                    if ((n = left.Count) != right.Count)
+                    {
+                        return false;
+                    }
+
+                    for (int i = 0; i < n; i++)
+                    {
+                        TValue? leftElem = left[i];
+                        TValue? rightElem = right[i];
+                        bool areEqual = leftElem is null ? rightElem is null : leftElem.Equals(rightElem);
+                        if (!areEqual)
+                        {
+                            return false;
+                        }
+                    }
+
+                    return true;
                 }
+            }
 
-                for (int i = 0; i < n; i++)
+            public int GetHashCode(JsonSerializerOptions options)
+            {
+                HashCode hc = default;
+
+                hc.Add(options._dictionaryKeyPolicy);
+                hc.Add(options._jsonPropertyNamingPolicy);
+                hc.Add(options._readCommentHandling);
+                hc.Add(options._referenceHandler);
+                hc.Add(options._encoder);
+                hc.Add(options._defaultIgnoreCondition);
+                hc.Add(options._numberHandling);
+                hc.Add(options._unknownTypeHandling);
+                hc.Add(options._defaultBufferSize);
+                hc.Add(options._maxDepth);
+                hc.Add(options._allowTrailingCommas);
+                hc.Add(options._ignoreNullValues);
+                hc.Add(options._ignoreReadOnlyProperties);
+                hc.Add(options._ignoreReadonlyFields);
+                hc.Add(options._includeFields);
+                hc.Add(options._propertyNameCaseInsensitive);
+                hc.Add(options._writeIndented);
+                hc.Add(options._typeInfoResolver);
```

Comment:
This is not in sync with the equals method. Equals method does referential equality and this calls the type provided hashcode that can be overridden to do whatever.

Reply:
Good point, I've updated both methods so that they always hardcode reference equality.